### PR TITLE
Initial draft of Disk Volume Mount Resources

### PR DESCRIPTION
**Pull request by lingmann**

### CODE REVIEW

**Comment by lingmann** [comment]
@mohitsoni and @jieyu could you review? Adding some documentation on how to use Mount Disk Resources with DC/OS and best practices for end-users.

**Comment by lingmann** [comment]
Ping @pyronicide @mohitsoni and @jieyu, would appreciate a quick review... thanks!

**Comment by jieyu** [comment]
LGTM!

File: 1.7/administration/storage.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+***Warning:*** This will terminate any running tasks or services on the node.
+
+1. Connect to an agent in the cluster with SSH
+
+2. Examine the current Mesos Agent resource state
+
+  Note there are no references yet for `/dcos/volume0`.
+
+	```
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: `bash`

File: 1.7/administration/storage.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	root@dcos-agent-public:~# cat /var/lib/dcos/mesos-resources
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: Let's scale this back to `#`

File: 1.7/administration/storage.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+	MESOS_RESOURCES='[{"ranges": {"range": [{"end": 21, "begin": 1}, {"end": 5050, "begin": 23}, {"end": 32000, "begin": 5052}]}, "type": "RANGES", "name": "ports"}, {"role": "*", "type": "SCALAR", "name": "disk", "scalar": {"value": 47540}}]'
+	```
+
+3. Stop the Mesos Agent
+
+  On a Private Agent:
+
+    ```bash
+	systemctl stop dcos-mesos-slave.service
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: `$`

File: 1.7/administration/storage.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	```
+
+  On a Public Agent:
+
+    ```bash
+	systemctl stop dcos-mesos-slave-public.service
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: `$`

File: 1.7/administration/storage.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+	```
+	root@dcos-agent-public:~# cat /var/lib/dcos/mesos-resources
+	# Generated by make_disk_resources.py on 2016-05-05 17:04:29.868595
+	#
+
+	MESOS_RESOURCES='[{"ranges": {"range": [{"end": 21, "begin": 1}, {"end": 5050, "begin": 23}, {"end": 32000, "begin": 5052}]}, "type": "RANGES", "name": "ports"}, {"role": "*", "type": "SCALAR", "name": "disk", "scalar": {"value": 47540}}]'
+	```
+
+3. Stop the Mesos Agent
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: Maybe combine this into a single command?

File: 1.7/administration/storage.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    ```bash
+	systemctl stop dcos-mesos-slave-public.service
+	```
+
+4. Clear Mesos Agent state
+
+    ```bash
+	# Remove Volume Mount Discovery resource state
+	rm -f /var/lib/dcos/mesos-resources
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: `$`

File: 1.7/administration/storage.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	```
+	
+	```bash
+	# Remove Mesos Agent checkpoint state
+	rm -f /var/lib/mesos/slave/meta/slaves/latest
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: `$`

File: 1.7/administration/storage.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+	```
+
+5. Create a 200MB loopback device
+
+  This is suitable for testing purposes only. Mount volumes must have at least 200MB of free space available. 100MB on each volume will be reserved by DC/OS and is not available for other services.
+
+    ```bash
+	mkdir -p /dcos/volume0
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: Prompt on the front please.

File: 1.7/administration/storage.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    ```bash
+	reboot
+	```
+
+8. SSH to Mesos Agent and Verify new resource state
+
+  Look through the journald logs for references to the new volume `/dcos/volume0`. In particular, there should be an entry for the Mesos Agent starting up and the new volume0 Disk Mount resource.
+
+    ```bash
+	root@dcos-agent-public:~# journalctl -b | grep '/dcos/volume0'
```
Comment (pyronicide) [inline 1.7/administration/storage.md]: Remove most of the prompt.

**Comment by lingmann** [comment]
@pyronicide I actually prefer to write commands in ways that people can easily copy paste and use '#' for inline comments... Do we have a strong convention here with our docs to include '#' or '$' at the beginning of commands? If not, I'd prefer to leave these commands as is.

**Comment by mohitsoni** [comment]
Looks good for dev use-case. Can we also expand this to cover a real-example with AWS EBS volumes ?

**Comment by gabrielhartmann** [comment]
@lingmann:  I agree with Mohit.  Also needs Azure examples.

**Comment by lingmann** [comment]
@mohitsoni @gabrielhartmann I think adding cloud specific examples is out of scope in this doc for the time being... it really explodes the complexity of the document and is very provider specific. How about I just add links to AWS and Azure specific volume management documentation and strategies instead?

**Comment by gabrielhartmann** [comment]
@lingmann:  Sounds good to me.  As long as a person can read this document and follow links and read more documents and get to a way of actually using MOUNT Volumes on DC/OS, I'm happy.

**Comment by pyronicide** [comment]
@lingmann we've got a pretty strong convention right now for `$` and `#` (every other doc should do it) to differentiate between shell commands and output. I share your frustration, wish there was an easier way to do it.

File: 1.7/administration/storage.md (hunk 5 above)
Comment (lingmann) [inline 1.7/administration/storage.md]: I thought about that, but we can't combine without making it into a little subscript or by swallowing systemctl errors.

**Comment by lingmann** [comment]
@pyronicide integrated all of your change requests. Please do a final pass and then ready for merge.

**Comment by lingmann** [comment]
Thanks @pyronicide! When will this be deployed to the live site? Want to follow up with a few folks who are looking for this info.

**Comment by pyronicide** [comment]
@lingmann 5m or so.

**Pull request merged**